### PR TITLE
ant plugin: switch to using ant snap for building (by default)

### PR DESCRIPTION
**Pull request by cjp256**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by cjp256** [comment]
@bodewig Hey Stefan, maybe you can take a look at these updates? :)

Tested ant hello world using:
(1) default, unspecified resulting in ant snap from latest/edge
(2) ant-version set to 1.10.5 with checksum
(3) ant-version set to 1.10.6 with wrong checksum (fails as expected)
(4) ant-snap-channel: "1.9/candidate"

**Comment by cjp256** [comment]
(rebased)

File: snapcraft/plugins/ant.py [hunk 1]
```diff
@@ -33,11 +33,16 @@
       (list of strings)
       Run the given ant targets.
 
+    - ant-snap-channel:
```
Comment (sergiusens) [inline snapcraft/plugins/ant.py]: can we call this `ant-channel`? Just to be consistent with all the other plugins.

Comment (cjp256) [reply snapcraft/plugins/ant.py]: Good call, fixed.

**Comment by cjp256** [comment]
(rebased with spread fix)

**Comment by bodewig** [comment]
Sorry for being late to the party. This looks good to me, and of course you know how to add a snap dependency while building better than me. :-)

You may want to make the doc for `ant-checksum` more explicit, but then again it already says it is the checksum for `ant-version` which already says it is an alternative to using snaps.

**Comment by bodewig** [comment]
BTW, you could have switched to `candiate` rather than `edge`, but I'll open a PR to make `stable/latest` the default once we've made that official - which I hope to be a question of days.

**Pull request merged**